ecosystem/python/sdk/aptos_sdk/transaction_worker.py

```diff
@@ -0,0 +1,225 @@
+
+
+class TransactionWorker:
+    """
+    The TransactionWorker provides a simple framework for receiving payloads to be processed. It
+    acquires new sequence numbers and calls into the callback to produce a signed transaction, and
+    then submits the transaction. In another task, it waits for resolution of the submission
+    process or get pre-execution validation error.
+
+    Note: This is not a particularly robust solution, as it lacks any framework to handle failed
+    transactions with functionality like retries or checking whether the framework is online.
+    This is the responsibility of a higher-level framework.
+    """
+
+    _account: Account
+    _account_sequence_number: AccountSequenceNumber
+    _rest_client: RestClient
+    _transaction_generator: typing.Callable[
+        [Account, int], typing.Awaitable[SignedTransaction]
+    ]
+    _started: bool
+    _stopped: bool
+    _outstanding_transactions: asyncio.Queue
+    _outstanding_transactions_task: typing.Optional[asyncio.Task]
+    _processed_transactions: asyncio.Queue
+    _process_transactions_task: typing.Optional[asyncio.Task]
+
+    def __init__(
+        self,
+        account: Account,
+        rest_client: RestClient,
+        transaction_generator: typing.Callable[
+            [Account, int], typing.Awaitable[SignedTransaction]
+        ],
+    ):
+        self._account = account
+        self._account_sequence_number = AccountSequenceNumber(
+            rest_client, account.address()
+        )
+        self._rest_client = rest_client
+        self._transaction_generator = transaction_generator
+
+        self._started = False
+        self._stopped = False
+        self._outstanding_transactions = asyncio.Queue()
+        self._processed_transactions = asyncio.Queue()
+
+    def account(self) -> AccountAddress:
+        return self._account.address()
+
+    async def _submit_transactions_task(self):
+        try:
+            while True:
+                sequence_number = (
+                    await self._account_sequence_number.next_sequence_number()
+                )
+                transaction = await self._transaction_generator(
+                    self._account, sequence_number
+                )
+                txn_hash_awaitable = self._rest_client.submit_bcs_transaction(
+                    transaction
+                )
+                await self._outstanding_transactions.put(
+                    (txn_hash_awaitable, sequence_number)
+                )
+        except asyncio.CancelledError:
+            return
+        except Exception as e:
+            # This is insufficient, if we hit this we either need to bail or resolve the potential errors
+            logging.error(e, exc_info=True)
+
+    async def _process_transactions_task(self):
+        try:
+            while True:
+                # Always start waiting for one
+                (
+                    txn_awaitable,
```

Comment:
haven't used SDK before, is this waiting for txn hash (polling the rest endpoint for txn hash) or what is it doing?

looking at ecosystem/python/sdk/aptos_sdk/async_client.py , I see that signature is 
    def submit_bcs_transaction(self, signed_transaction: SignedTransaction) -> str:
so return type is str, not something you can wait on?